frontend/src/components/PredictionTools.jsx

```diff
@@ -75,79 +90,88 @@ const PredictionTools = ({ portfolioData, companiesData, portfolioValue }) => {
   useEffect(() => {
     setPredictionData(null);
     setPredictedBalance(null);
+    getModelExists();
   }, [portfolioValue]);
 
   return (
-    <div className="bg-white h-150 mt-20 rounded-xl ml-30">
+    <div className="bg-white h-150 mt-20 rounded-xl ml-30 min-w-[55%]">
       <h2 className="text-3xl p-3 text-center w-full">Analysis Tool</h2>
       <div className="flex flex-row w-full h-full items-center -mt-12">
-        <div className="flex flex-col justify-center">
-          <button
-            className=" bg-green-700 text-white hover:brightness-80 mx-5 m-2 hover:scale-105"
-            onClick={() => getModel(false)}
-          >
-            Get Future Predictions
-          </button>
-          <button className=" bg-red-700 text-white hover:brightness-80 mx-5 m-2 hover:scale-105">
-            Risk Analysis
-          </button>
-          {predictionsClicked == false && (
-            <NewModelButton getModel={getModel} />
-          )}
-          {predictionsClicked == true && (
-            <>
-              <div className="mr-auto ml-auto rounded-full w-8 h-8 m-3 border-3 border-t-transparent border-orange-200 animate-spin"></div>
-              <h2 className="w-50 mr-auto ml-auto pl-3 pr-2">
-                Predictions take time, and are computationally demanding...
-              </h2>
-            </>
-          )}
-          <h2 className="font-bold mr-auto ml-auto mt-10">
-            Total Current Balance
-          </h2>
-          {companiesData !== null && (
-            <h2 className="font-bold mr-auto ml-auto predictedBalance">
-              {portfolioValue}
-            </h2>
-          )}
-          {companiesData == null && (
-            <h2 className="font-bold mr-auto ml-auto text-gray-800">
-              loading...
-            </h2>
-          )}
-          {predictedBalance != null && (
-            <>
+        {companiesData != null && companiesData.length !== 0 && (
+          <>
+            <div className="flex flex-col justify-center">
+              {isModelExists && (
+                <>
+                  <button
+                    className=" bg-green-700 text-white hover:brightness-80 mx-5 m-2 hover:scale-105"
+                    onClick={() => getModel(false)}
+                  >
+                    Get Future Predictions
```

Comment:
This would show the button if a model exists, but user still needs to click to load it? Would be nice to load automatically.